Formula/sapling.rb

```diff
@@ -22,13 +22,13 @@ class Sapling < Formula
   depends_on "gh"
   depends_on "node"
   depends_on "openssl@1.1"
-  depends_on "python@3.10"
+  depends_on "python@3.11"
 
   def install
     ENV["OPENSSL_DIR"] = Formula["openssl@1.1"].opt_prefix
-    ENV["PYTHON_SYS_EXECUTABLE"] = Formula["python@3.10"].opt_prefix/"bin/python3.10"
-    ENV["PYTHON"] = Formula["python@3.10"].opt_prefix/"bin/python3.10"
-    ENV["PYTHON3"] = Formula["python@3.10"].opt_prefix/"bin/python3.10"
+    ENV["PYTHON_SYS_EXECUTABLE"] = Formula["python@3.11"].opt_prefix/"bin/python3.11"
+    ENV["PYTHON"] = Formula["python@3.11"].opt_prefix/"bin/python3.11"
+    ENV["PYTHON3"] = Formula["python@3.11"].opt_prefix/"bin/python3.11"
```

Comment:
I don't think these environment variables do anything. The Makefile uses assignment operators, e.g. `PYTHON := python2` so the values aren't overridden. They would need to be passed in as arguments, `make PYTHON=... PYTHON3=...`

In current formula, the value of `PYTHON3` is `python3`. This is then used in script https://github.com/facebook/sapling/blob/0.2.20221222-152408-ha6a66d09/eden/scm/contrib/pick_python.py#L24-L26 which prioritizes some arbitrary Pythons (3.8, 3.10, 3.7, 3.6)